graylog2-server/src/main/java/org/graylog2/rest/resources/system/DeflectorResource.java

```diff
@@ -70,10 +70,12 @@ public DeflectorResource(IndexSetRegistry indexSetRegistry,
     @Produces(MediaType.APPLICATION_JSON)
     @Deprecated
     public DeflectorSummary deprecatedDeflector() throws TooManyAliasesException {
-        final IndexSet indexSet = indexSetRegistry.getDefault()
-                .orElseThrow(() -> new NotFoundException("Default index set not found"));
-
-        return DeflectorSummary.create(indexSet.isUp(), indexSet.getCurrentActualTargetIndex());
+        try {
+            final IndexSet indexSet = indexSetRegistry.getDefault();
+            return DeflectorSummary.create(indexSet.isUp(), indexSet.getCurrentActualTargetIndex());
+        } catch (IllegalStateException e) {
+            throw new NotFoundException("Default index set not found");
```

Comment:
I would throw an `InternalServerErrorException` here (or simply let the `IllegalStateException` bubble up which results in the same) because it's actually an error in the Graylog system, not a wrong request by the client.